src/content/guides/creating-your-own-admission-controller/index.md

```diff
@@ -0,0 +1,219 @@
+
+__Note:__ I would like to mention that Kubernetes API allows you to register custom resource objects too, called `Custom Resource Definition`. In addition, you can create an API extension server which listens to a new REST path. But in this guide, we focus exclusively on the admission functionality.
+
+## Goal
+
+Our goal here is to create a simple validation controller which will empower us to influence the pod creation. Although there are many more possibilities and the logic can be as complex as you need. The goal is to create a basic version which makes a simple validation. You can find more real examples in the links at the bottom.
+
+Our example controller will be called `grumpy` and will reject all new pods with a name different than `smooth-app`. I recognize it may be tempting to deploy this controller in a real cluster ;).
+
+## How API proxies requests
+
+The Kubernetes API server needs to know when to send the incoming request to our admission controller. The Kubernetes nature advocates the use of a declarative state always and this is no exception. Below we define a `ValidationWebhookConfiguration` which gives the needed information to the API:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: grumpy
+webhooks:
+  - name: grumpy.giantswarm.io
+    clientConfig:
+      service:
+        name: grumpy
+        namespace: default
+        path: "/validate"
+      caBundle: "${CA_BUNDLE}"
+    rules:
+      - operations: ["CREATE"]
+        apiGroups: [""]
+        apiVersions: ["v1"]
+        resources: ["pods"]
+```
+
+The configuration definition is pretty obvious. You can see there are two main parts to be considered. In the first one, `clientConfig`, we define where our service can be found (can be an external URL too), and the `path` which our validation server will listen. Also, you notice there is a `CA` to be defined. Since security is always important, adding the cert authority will tell Kubernetes API to use HTTPS and validate our server using the passed asset. In the next section, you will find an explanation on how to generate all the certs needed.
+
+The second part specifies which rules will follow the API to decide if a request should be forwarded for validation to `grumpy` or not. Here it is configured that only requests with method equal to `CREATE` and resource type `pod` will be forwarded. 
+
+## Generate the certificates and the CA
+
+Since the scope of this guide is not teaching how to build a PKI bundle, we have created a script, `gen_cert.sh`, in the grumpy repository which generates a CA bundle and a key pair for our grumpy server. Also we need to provide the CA in the webhook displayed above, in order to allow the Kubernetes API to create a secure connection against our shiny controller.
```

Comment:
link to repo here

Reply:
```suggestion
Since the scope of this guide is not teaching how to build a PKI bundle, we have created a script, `gen_cert.sh`, in the [grumpy repository](https://github.com/giantswarm/grumpy) which generates a CA bundle and a key pair for our grumpy server. Also we need to provide the CA in the webhook displayed above, in order to allow the Kubernetes API to create a secure connection against our shiny controller.
```